### DISCUSSION TITLE
How to implement a dropdown with country flags

### DISCUSSION THREAD

**Discussion by amyzheng95** [twilio-labs/paste#671] [Q&A] [answered]
![Screen Shot 2020-08-19 at 4 00 03 PM](https://user-images.githubusercontent.com/15934761/90777105-d477da00-e2af-11ea-91d1-7d77eb5bc0a8.png)

Was looking at using Paste Select but was wondering if I'll have to import all the flag icons to make it work?

**Comment by serifluous** [comment]
Thanks, @amyzheng95 ! We've been discussing with our team how to do this with Paste components. Right now neither Select nor [Combobox](https://paste.twilio.design/components/combobox) can take images, so we're discussing some solutions.

Do you have a page screenshot/example of where you're planning on using this?

**Comment by serifluous** [comment, marked as answer]
@amyzheng95 Since the Select and Combobox components can't take images right now, based on the design mockup you shared with us on Slack, you can:

1. Replace the flag with the country name and country code (e.g., "Estonia (+372)")
2. Remove the country code prefix on the form input next to it.

I definitely recommend chatting with the product designer on your team, though, to check if there are any other possible solutions.

**Comment by vankhoawin** [comment]
Revisiting this: now that Combobox can take a `optionTemplate` prop to render custom JSX, is there any interest in adding country icons in order for us to create a Country Dropdown using Paste?

https://paste.twilio.design/components/combobox#component-props

**Reply by richbachman** [reply]
Hi @vnguyen94. We've discussed ideas about how to accomplish this with the best developer experience in mind, i.e. what's the best way to import all the country icons. But nothing is set in motion yet.

We're open to hearing your thoughts, so let us know if you have any solutions.

**Reply by vankhoawin** [reply]
hmm interesting lol. what if Paste exported all of the country icon SVGs as usual. A CountryDropdown component (which doesn't necessarily have to live in Paste) could take those SVGs and combine them into a spritesheet/icon font?

A CountryDropdown doesn't necessarily need all the pros of SVGs and might be better suited with a sprite sheet:
- CountryDropdown icons will be in a fixed dimension
- CountryDropdown icons don't need any manipulations e.g. background color change
- spritesheet should be smaller in size in comparison to inline SVGs, can be cached, and don't need to be parsed

cons:
- SVGs are supposedly more a11y-friendly? i haven't tried personally

i'm just spitting ideas if it wasn't readily apparent 😄  but support for a CountryDropdown would really help out a lot of Twilio teams

**Reply by vankhoawin** [reply]
just to tie this up, our team's CountryDropdown component is now using Unicode for country flags, which is well-supported by browsers and doesn't have a bundle size impact as any of the above options do.

**Comment by ghelton** [comment]
We currently have a design that allows users to select a status for their device. This design displays a supporting icon that we display alongside the label. We would also like this to be displayed when selected.

Ideal display:
![Screen Shot 2021-11-15 at 8 29 36 AM](https://user-images.githubusercontent.com/48418/141818325-5cec00cf-637e-4263-93b8-5ee78c71929c.png)

What we have currently:
<img width="308" alt="Screenshot 2021-11-15 at 18 28 57" src="https://user-images.githubusercontent.com/48418/141818368-efba4bfd-1599-4a2b-aa38-97d4762a46f2.png">

Ideally we'd add a Selected Item Template or something along those lines, currently only a string is supported.
This came up once before and we worked around it by not displaying the icon. Any chance we can reconsider this functionality? Is there room for us to add support ourselves in the future if we'd like?

**Reply by gloriliale** [reply]
Hi @ghelton! Right now, we're not planning on adding a way to display anything but strings when an option is selected in Combobox. There's two things that might be good options though: using a prefix for combobox for the status icons, or using our Combobox Primitive to create something that operates in the specific way you're looking for.

**Reply by ghelton** [reply]
Ok can you elaborate on the reasoning here? It seems sensible that the content you select match the content displayed.

Is this something our team can contribute if there's no reasoning against it? I'm trying to avoid maintaining a custom combo box if possible.

**Reply by gloriliale** [reply]
Hi @ghelton, I'm getting a better idea of the feasibility of this from the team, let me get back to you!

**Reply by gloriliale** [reply]
Hi @ghelton! Got the green light from design; we can move forward with this as a contribution! Do you have a time frame you were thinking about doing this work in? It'd be great to get this in front of committee as well; we can do that either before a PR or after, depending on your timeline. This is a new type of contribution so I'm excited to see how we can make it happen!

**Reply by ghelton** [reply]
Great! I think I might try and spend some time on this during Tweek 🤞🏻

**Reply by gloriliale** [reply]
Hi again @ghelton!
I got some clarification on this in particular, and it looks like it's not possible foundationally 😞 .  We use native HTML input elements such as `<select />` and `<input />` to build our components, including Combobox,, and these HTML elements don't allow for images, icons, or svgs to be added to the value of the control. The best way we've found to get this kind of functionality would be what Van posted above: his team's solution is using unicode emojis. Thanks for bearing through so much back and forth on this.
